### ISSUE TITLE
[NTI] IArrayLike: Cannot do '[]' access on a struct

### ISSUE THREAD

**Issue by Dominator008** [google/closure-compiler#1531]
Repro:

```js
/** @param {!IArrayLike<string>} x */
function f(x) {
  return x[0];
}
```

http://closure-compiler-debugger.appspot.com/#input0%3D%252F**%2520%2540param%2520%257B!IArrayLike%253Cstring%253E%257D%2520x%2520*%252F%250Afunction%2520f(x)%2520%257B%250A%2520%2520return%2520x%255B0%255D%253B%250A%257D%26input1%26conformanceConfig%26externs%26refasterjs-template%26includeDefaultExterns%3D1%26CHECK_SYMBOLS%3D1%26CHECK_TYPES%3D1%26CHECK_TYPES_NEW_INFERENCE%3D1%26LANG_IN_IS_ES6%3D1%26MISSING_PROPERTIES%3D1%26PRESERVE_TYPE_ANNOTATIONS%3D1%26PRETTY_PRINT%3D1%26TRANSPILE%3D1

Real-world example: https://github.com/google/closure-library/blob/master/closure/goog/array/array.js#L505

**Comment by MatrixFrog** [comment]
Oh right, because interfaces are always `@struct` in the NTI. We'll need to make an exception in this case :)

**Comment by concavelenz** [comment]
It is still @struct NTI just need to cognize IObject as  as the [] "operator"

**Comment by dimvar** [comment]
I'll work on handling IObject/IArrayLike in NTI this week.

**Comment by Dominator008** [comment]
Closed by b0b676c18657f6d8590c79de47c625e170ebae3d.
